### ISSUE TITLE
Devel: Can't stay authenticated 

### ISSUE THREAD

**Issue by lduboeuf** [ubports/morph-browser#347]
in apps that use the Morph.Web component, we can't stay connected.
Tested on Gmail, Youtube, Twitter webapps
Is there anything to be done ?

**Comment by balcy** [comment]
I think the profile of the new QtWebEngine (https://doc.qt.io/qt-5/qml-qtwebengine-webengineprofile.html) has now `offTheRecord:true` as default. 
https://bugreports.qt.io/browse/QTBUG-79685
So setting `context.offTheRecord:true` in the MorphWebView should solve it. Not sure if we can / should sth. globally, because incognito by default seems better privacy wise.

**Comment by mateosalta** [comment]
nothing is staying signed in in the browser now...

**Comment by balcy** [comment]
gitlab / github stays logged in for me in morph, and github in the webapp, too.

**Comment by mretallack** [comment]
Hi,
Is there a run-time work around for setting offTheRecord to false?

**Comment by mretallack** [comment]
I have found that setting offTheRecord to False in MorphWebContext allows the cookies to persist. I have no idea if this is the correct solution.

 I have attached a Patch


[fix-offTheRecord.patch.txt](https://github.com/ubports/morph-browser/files/4975363/fix-offTheRecord.patch.txt)

Update: This may only effect PinePhone, https://github.com/ubports/morph-browser/blob/bb9f86efd727f100e1cc60390db0955521c02240/src/Morph/Web/MorphSharedWebContext.qml#L27

**Comment by balcy** [comment]
yes, but the current xenial (DEV branch) version does include this change already.

https://github.com/ubports/morph-browser/blob/xenial/src/Morph/Web/MorphSharedWebContext.qml

Is it possible morph-browser does not use the xenial branch on some phones / channels (e.g. pinephone) ?

**Comment by balcy** [comment]
oh there are https://github.com/ubports/morph-browser/tree/xenial_-_edge and https://github.com/ubports/morph-browser/tree/xenial_-_edge_-_pine branches...

**Comment by balcy** [comment]
in the QA group I've seen Martin had the following in the log: `qml: the temporary default profile does no longer exit`
Therefore I've changed the way the temporary profile (to get the chromium version) is created.
It would not explain that no sessions at all are saved though.
https://github.com/ubports/morph-browser/pull/359

**Comment by balcy** [comment]
Is it still the case that the one can't stay authenticated with the current dev version ?

**Comment by lduboeuf** [comment]
Just tested with Mateo's Youtube webapp and github webapp, and still not able to get authenticated after reload of app.I've tried also by cleaning cache of the app. Still the same

But why todays update ?. Last commit on xenial is 3 days ago

**Comment by balcy** [comment]
Yes I thought it had to do sth. with https://github.com/ubports/morph-browser/commit/e66f81455f4f0ac80eda340bc3fa45831c7be1a8 but that was not the case.

The app https://open-store.io/app/github.rubencarneiro uses QtWebEngine directly, and would have to be rebuilt with `offTheRecord: false` added to the `WebEngineProfile` (the source link https://gitlab.com/rubencarneiro/webgithub is no longer available though)

The youtube app uses Morph.Web, which takes the default for `offTheRecord` (or `incognito` as alias) from WebEngineProfile. Right now `MorphWebContext` (https://github.com/ubports/morph-browser/blob/xenial/src/Morph/Web/MorphWebContext.qml) used by apps would default to incognito:true / offTheRecord:true if not explicitly set.
So adding [edit:correction] `offTheRecord:false` to `WebContext` in the webapp would solve it.

@UniversalSuperBox  Do you think we should make `offTheRecord:false` the default in `WebContext` of Morph.Web ? 
This would provide compatibility with existing apps, but no longer have the same default as `WebEngineProfile`.
E.g. a developer moving from Morph.Web to QtWebEngine later would have a different effect for profiles. Privacy wise (and also memory) it is good to have memory-only as default and only change it to disk storage if needed. We might even introduce a command line flag for webapp-container so that only apps that need to save the session switch to this mode.

**Comment by balcy** [comment]
... I can write PRs for the most important affected webapps.
The line can safely be added now and the app will be prepared for OTA13

**Comment by lduboeuf** [comment]
@balcy i'm voting for the upgrade web apps solution, just need to broadcast developpers about it. But just afraid with the holidays season in EU...

**Comment by mateosalta** [comment]
> Just tested with Mateo's Youtube webapp and github webapp, and still not able to get authenticated after reload of app.I've tried also by cleaning cache of the app. Still the same
> 
> But why todays update ?. Last commit on xenial is 3 days ago

youtube(one from me) in the store doesn't use morph yet ;p - the github version is what we are wroking on but has the breaking change for dropdown menus, so no off the record fix yet, i will try and get that in

**Comment by UniversalSuperBox** [comment]
My vote says that anything that changes behavior should be avoided. Luckily QML versioning could save us here. For this release (and the current Morph.Web import) we should make the changes needed to have no change in behavior. We can bump the version of the Morph.Web import to provide a new default and include that with a new Ubuntu Touch framework version so apps can base against it in the future.

In other words: Keep behavior the same as before. Set offTheRecord = false by default.

**Comment by UniversalSuperBox** [comment]
Okay, further inspection reveals the information that was already in this thread but not necessarily acted on:

Qt changed the default for offTheRecord from `false` to `true` in QtWebEngine 5.13. This means that anyone that used a WebEngineProfile and **did not set offTheRecord** saw a behavior change.

MorphWebView will, by default, use one of the [MorphSharedWebContext](https://github.com/ubports/morph-browser/blob/xenial/src/Morph/Web/MorphSharedWebContext.qml)s when one isn't provided. Which is used depends on whether the `incognito` property is set. So that's fine.

We will set `offTheRecord: false` in [MorphWebContext](https://github.com/ubports/morph-browser/blob/xenial/src/Morph/Web/MorphWebContext.qml) which will cause the default to persist for anyone instancing a MorphWebContext. But for the apps like [vin4ter's Gmail webapp](https://open-store.io/app/gmail.vin4ter) which create a WebEngineProfile and not a MorphWebContext, there's really nothing we can do for them...

**Comment by UniversalSuperBox** [comment]
Well, we could patch QtWebEngine. But that seems like a bit of a fool's errand. Qt really should have made the default change part of a QML versioning change, but hasn't for some reason.

**Comment by UniversalSuperBox** [comment]
In other news, I am a fool and this was my errand, heh. The patch is proposed for setting offTheRecord to false globally. We'll need to discuss removing it appropriately after the OTA-13 release, if it is merged.

**Comment by UniversalSuperBox** [comment]
I've tested the GitHub webapp today and it saves cookies.

**Comment by sap-nocops** [comment]
Works on my OPO latest dev, github webapp!

**Comment by cibersheep** [comment]
Works on Morph signing in in GitLab.
E5HD RC channel
